webapp/channels/src/components/common/hooks/use_desktop_notification_permission.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+
+import {type NotificationPermissionNeverGranted} from 'utils/notifications';
+import {isNotificationAPISupported} from 'utils/notifications';
+import {isDesktopApp} from 'utils/user_agent';
+
+type DesktopNotificationPermission = Exclude<NotificationPermission, typeof NotificationPermissionNeverGranted> | undefined;
+
+// This is a hook that is used to request notification permission for desktop app
+// it also returns the permission state. We use this as a workaround for Bug with Electron - https://github.com/electron/electron/issues/11221
+// tl;dr Electron always show 'granted' when queries for Notification.permission, hence this workaround
+export function useDesktopNotificationPermission() {
```

Comment:
Nit, I think this should be `useDesktopAppNotificationPermission` since the web app still uses desktop notifications

Reply:
changed